### PR TITLE
Split form behavior

### PR DESCRIPTION
**Pull request by itkg-nanne**
[OO-FEATURE] allow to extend js form's behavior

### CODE REVIEW

**Comment by itkg-nanne** [comment]
ok for this, @itkg-alavieille ?

File: BackofficeBundle/Resources/public/coffee/formBehavior/FormLibraryBehavior.coffee [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  ###*
+   * activateBehaviors
+   * @param {Object} view
+   * @param {Object} form
+  ###
+  activateBehaviors: (view, form) ->
+    for formBehavior in @formBehaviors
+      formBehavior.activateBehavior(view, form)
+
+formBehaviorLibrary = new OpenOrchestra.FormBehavior.FormBehaviorLibrary()
```
Comment (itkg-alavieille) [inline BackofficeBundle/Resources/public/coffee/formBehavior/FormLibraryBehavior.coffee]: can you add it in the namespace ?

File: BackofficeBundle/Resources/public/coffee/formBehavior/ColorPicker.coffee [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+   * @param {Array} elements
+   * @param {Object} view
+   * @param {Object} form
+  ###
+  activateBehaviorOnElements: (elements, view, form) ->
+    for element in elements
+      element = $(element)
+      element.minicolors()
+
+formBehaviorLibrary.add(new OpenOrchestra.FormBehavior.ColorPicker(".colorpicker"))
```
Comment (itkg-alavieille) [inline BackofficeBundle/Resources/public/coffee/formBehavior/ColorPicker.coffee]: use 
 jQuery ->

**Comment by itkg-nanne** [comment]
changes done @itkg-alavieille , @itkg-ngilain

File: BackofficeBundle/Resources/public/coffee/formBehavior/Token.coffee [hunk 3]
```diff
@@ -0,0 +1,107 @@
+###*
+ * @namespace OpenOrchestra:FormBehavior
+###
+window.OpenOrchestra or= {}
+window.OpenOrchestra.FormBehavior or= {}
+
+###*
+ * @class Token
+###
+class OpenOrchestra.FormBehavior.Token extends OpenOrchestra.FormBehavior.AbstractFormBehavior
```
Comment (itkg-ngilain) [inline BackofficeBundle/Resources/public/coffee/formBehavior/Token.coffee]: ``Token`` is not full of sense, can you rename it into something more oriented on what its doing ? (I don't get the context of what this behavior add to a form with that name)

File: BackofficeBundle/Resources/public/coffee/orchestraLib.coffee [hunk 4]
```diff
@@ -121,205 +121,15 @@ smartConfirm = (logo, titleColorized, text, functions) ->
...
-  activateSelect2(elements) if (elements = $(".select2", form)) && elements.length > 0
-  activateToken(elements) if (elements = $(".select-boolean", form)) && elements.length > 0
-  activateOrchestraNodeChoice(elements) if (elements = $(".orchestra-node-choice", form)) && elements.length > 0
-  activateColorPicker(elements) if (elements = $(".colorpicker", view.el)) && elements.length > 0
-  activateHelper(elements) if (elements = $(".helper-block", form)) && elements.length > 0
-  activateTinyMce(view, elements) if (elements = $("textarea.tinymce", form)) && elements.length > 0
-  activateHidden(elements) if (elements = $("input[type='hidden'][required='required']", form)) && elements.length > 0
-  activateDatepicker(elements) if (elements = $(".datepicker", form)) && elements.length > 0
-  ajaxRefreshForm(elements, view, form) if (elements = $(".refresh-form", form)) && elements.length > 0
+  try
```
Comment (itkg-alavieille) [inline BackofficeBundle/Resources/public/coffee/orchestraLib.coffee]: why try ?

Comment (itkg-nanne) [inline BackofficeBundle/Resources/public/coffee/orchestraLib.coffee]: because need three levels of typeof, @itkg-alavieille

**Comment by itkg-nanne** [comment]
rename ok for you, @itkg-ngilain ?

File: BackofficeBundle/Resources/public/coffee/orchestraLib.coffee (hunk 4 above)
Comment (itkg-ngilain) [inline BackofficeBundle/Resources/public/coffee/orchestraLib.coffee]: I agree with @itkg-alavieille : with that try catch, a little error somewhere freeze the entire process.

Plus it don't give any clue to the dev of what is causing the problem

**Comment by itkg-ngilain** [comment]
I'm ok with the renaming !
But I also agree with @itkg-alavieille with the too hard try catch

**Comment by itkg-nanne** [comment]
changes done @itkg-ngilain , @itkg-alavieille

**Comment by itkg-alavieille** [comment]
:+1:

**Pull request merged**